src/DynamoCoreWpf/ViewModels/Core/OutPortViewModel.cs

```diff
@@ -294,8 +305,14 @@ private void OnMouseLeftButtonDownOnContext(object parameter)
         }
 
         protected override void RefreshPortColors()
-        {            
-            if (node.NodeModel.IsPartiallyApplied)
+        { 
+            //This variable checks if the node is a function class
+            bool isFunctionNode = node.NodeModel.CachedValue != null &&
```

Comment:
you should put this somewhere you can share it - but lets find something better to replace it with.

Reply:
@aparajit-pratap do you know of any nice ways to check if the `NodeModel.CachedValue and its underlying stackValue` is a DS Function?

It appears that `StackValue` actually has a `IsFunctionPointer` and `IsFunctionIndex` method - would those be useful to expose in MirrorData as a public method - like `IsFunction`?

https://github.com/DynamoDS/Dynamo/blob/master/src/Engine/ProtoCore/Reflection/MirrorData.cs#L275

What is the actual object type when we see a `Function` in the preview bubble, is it a FunctionObject class?

Reply:
The actual object is a DesignScript class called `Function`, which is defined in `FunctionObject.ds`. The stackvalue will most likely be an object pointer.

Reply:
Awesome, I will add that in the next commit. Also, I think we can remove the `isReturningDefaultValue ` boolean as it's looking quite confusing and appears to have the same objective.

Reply:
Same here, shouldn't this be:
```
bool isFunctionNode = node.NodeModel.IsPartiallyApplied || node.NodeModel.CachedValue != null && node.NodeModel.CachedValue.IsFunction;
```